bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/internal/ScriptEngineManagerImpl.java

```diff
@@ -62,63 +59,73 @@ public void unsetScriptExtensionManager(ScriptExtensionManager scriptExtensionMa
     }
 
     @Reference(cardinality = ReferenceCardinality.MULTIPLE, policy = ReferencePolicy.DYNAMIC)
-    public void addScriptEngineFactory(ScriptEngineFactory provider) {
-        this.scriptEngineFactories.add(provider);
-
-        for (String language : provider.getLanguages()) {
-            this.supportedLanguages.put(language, provider);
+    public void addScriptEngineFactory(ScriptEngineFactory engineFactory) {
+        engineFactory.logFactoryDetails();
```

Comment:
I don't think we whould handle it this way.

If I would like to see the factory details of the added script engine factories in the log, I don't know how the logger needs to be configured.

The factory logs the details, so I don't know for which package I need to set the log level.
I also don't know which log level is used at all.
The factory A will e.g. use trace, the factory B will use debug another one info, ...

If we need that factory details for further information in the log at all, the `ScriptEngineFactory` should define a method that returns the factory details (e.g. a string).

Then this class (the engine manager) could define the log level. It will be equal for every implementation.
And we knows for which package we need to set the log level.

Reply:
> If I would like to see the factory details of the added script engine factories in the log, I don't know how the logger needs to be configured.
> 
> The factory logs the details, so I don't know for which package I need to set the log level.
> I also don't know which log level is used at all.
> The factory A will e.g. use trace, the factory B will use debug another one info, ...

I don't follow your logic. These logs will all be coming in from under org.openhab.core.automation.module.script, so set the log level there. But even this is too granular to see much going on, so I usualy have org.openhab.core.automation set to DEBUG or TRACE.

As per the discussion about getScriptTypes(), I have moved this logging out of the SEFs and into the SEMImpl.